src/neural_data_simulator/core/ephys_generator.py

```diff
@@ -818,18 +704,15 @@ def _execute(self):
 
         self._stream_continuous_data(continuous_data)
         self._stream_lfp(continuous_data)
-        self._stream_spike_events(spike_events, n_samples, time_elapsed)
+        self._stream_spike_events(spike_events)
 
     def _stream_continuous_data(self, continuous_data: ndarray):
         if self._outputs.raw is not None:
-            self._outputs.raw.send_as_chunk(continuous_data / self._resolution)
-
-    def _stream_spike_events(
-        self, spike_events: SpikeEvents, n_samples: int, time_elapsed: float
-    ):
-        if self._last_output_time is None:
-            raise ValueError("Last output time is not set.")
+            self._outputs.raw.send_array(
+                continuous_data / self._resolution, timestamps=None
+            )
```

Comment:
Another option would to explicitly use the `Timer` timestamps to pass the `timestamps` in (instead of implicitly letting LSL determine the timestamps underneath)